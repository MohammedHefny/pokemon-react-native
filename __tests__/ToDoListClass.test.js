import React from 'react';
import renderer from 'react-test-renderer';
import ToDoListClass from '../components/ToDoList/ToDoListClassFunc';

test('renders correctly', () => {
  const tree = renderer.create(<ToDoListClass />).toJSON();
  expect(tree).toMatchSnapshot();
});

// // ToDoListClass.test.tsx
// import React from 'react';
// import {render, fireEvent} from '@testing-library/react-native';
// import ToDoListClass from './ToDoListClass';

// describe('ToDoListClass Component', () => {
//   it('should render the input and button correctly', () => {
//     const {getByPlaceholderText, getByText} = render(<ToDoListClass />);

//     expect(getByPlaceholderText('Please type Your Task Name')).toBeTruthy();
//     expect(getByText('Add Task')).toBeTruthy();
//   });

//   it('should add a new task when the Add Task button is pressed', () => {
//     const {getByPlaceholderText, getByText, getByTestId} = render(
//       <ToDoListClass />,
//     );

//     const input = getByPlaceholderText('Please type Your Task Name');
//     const button = getByText('Add Task');

//     // Simulate user typing in the TextInput
//     fireEvent.changeText(input, 'New Task');

//     // Simulate button press
//     fireEvent.press(button);

//     // Check if the new task is added to the list
//     expect(getByText('New Task')).toBeTruthy();
//   });

//   it('should delete a task when the Delete button is pressed', () => {
//     const {getByText, getAllByText} = render(<ToDoListClass />);

//     // Find the Delete button for the first task and press it
//     const deleteButton = getAllByText('Delete')[0];
//     fireEvent.press(deleteButton);

//     // Ensure the task is removed from the list
//     expect(getByText('work')).not.toBeTruthy();
//   });

//   it('should toggle task completion when the Complete/Undo button is pressed', () => {
//     const {getByText, getAllByText} = render(<ToDoListClass />);

//     // Find the Complete button for the first task and press it
//     const completeButton = getAllByText('Complete')[0];
//     fireEvent.press(completeButton);

//     // Check if the task is marked as completed
//     const task = getByText('work');
//     expect(task.props.style).toContainEqual(
//       expect.objectContaining({textDecorationLine: 'line-through'}),
//     );
//   });
// });
