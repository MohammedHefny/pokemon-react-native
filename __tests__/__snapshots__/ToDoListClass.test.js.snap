// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <TextInput
    autoCapitalize="words"
    autoCorrect={true}
    onChangeText={[Function]}
    placeholder="Please type Your Task Name"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "plum",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginVertical": 10,
        "padding": 16,
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Task
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginVertical": 10,
      }
    }
  >
    Your Tasks Using Class Component
  </Text>
  <RCTScrollView
    data={
      [
        {
          "completed": false,
          "id": 0,
          "task": "work",
        },
        {
          "completed": false,
          "id": 1,
          "task": "study",
        },
        {
          "completed": false,
          "id": 2,
          "task": "play",
        },
        {
          "completed": false,
          "id": 3,
          "task": "sleep",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            work
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                  },
                  false,
                ]
              }
            >
              Complete
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Edit
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            study
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                  },
                  false,
                ]
              }
            >
              Complete
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Edit
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            play
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                  },
                  false,
                ]
              }
            >
              Complete
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Edit
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginVertical": 5,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "flex": 1,
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            sleep
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                  },
                  false,
                ]
              }
            >
              Complete
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Edit
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#007bff",
                "borderRadius": 4,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
